tardis/plasma/properties/nlte_rate_equation_solver.py

```diff
@@ -573,13 +643,12 @@ def prepare_first_guess(
         numpy.array
             Guess for ion number densities and electron density.
         """
-        # TODO needs to be changed for excitation
-        array_size = (number_density.index.values + 1).sum() + 1
-        first_guess = np.zeros(array_size)
-        index = 1
+        first_guess = pd.Series(0.0, index=rate_matrix_index)
         for atomic_number in atomic_numbers:
-            first_guess[index] = number_density.loc[atomic_number]
-            index += atomic_number + 1
+            first_guess.loc[(atomic_number, 1)][0] = number_density.loc[
```

Comment:
In principle, after the first iteration we should use the level number densities from the previous iteration as a first guess. For now, maybe adding a TODO note is sufficient.